rfc-091-sharing-assets.md

```diff
@@ -0,0 +1,43 @@
+# Sharing assets
+
+## Summary
+
+We need to share assets to make sure users only have to download a thing once. To do this, we'll upload each application's assets to the same location.
+
+## Problem
+
+In [RFC84 we proposed replacing Static with a gem](https://github.com/alphagov/govuk-rfcs/pull/84). One of the architectural challenges with this approach is that by moving assets (like CSS, JS and images) into the gem, the assets won't be shared for users. This means that users visiting GOV.UK would have to download assets multiple times, depending on what app renders the page.
+
+## Proposal
+
+We upload the assets to a central location. For example, if both `collections` and `government-frontend` have a file named `foo.css`, they will upload the file to `assets.publishing.service.gov.uk/shared/foo.css`. Because [Rails fingerprints assets](http://guides.rubyonrails.org/asset_pipeline.html#what-is-fingerprinting-and-why-should-i-care-questionmark), the asset actually uploaded will be `assets.publishing.service.gov.uk/shared/foo-8d811b8c3badbc0b0e2f6e25d3660a96cc0cca7993e6f32e98785f205fc40907.css`, unless the files are different, in which case they'll have different fingerprints and won't overwrite.
```

Comment:
> For example, if both `collections` and `government-frontend` have a file named `foo.css`

As part of this solution we'd need a way to ensure or encourage different frontend apps to use assets in a way that is shareable.

For example, `application.scss` in government-frontend might include a component with some application styles, `application.scss` in collections, could do the same – the same component and its own application styles. That leads to duplication of CSS that users would download twice.